### PR TITLE
Fix core tests

### PR DESCRIPTION
**Pull request by RMadjev**
Add __init__.py in tcms/core/tests and fix all failing tests

### CODE REVIEW

File: tcms/core/tests/test_admin.py [hunk 1]
```diff
@@ -44,21 +43,22 @@ def test_admin_display(self):
...
         self.assertNotContains(response, 'Test case status')
         self.assertNotContains(response, 'Test case texts')
-        self.assertNotContains(response, 'Test cases')
+        self.assertContains(response, 'Search Test Cases')
 
         # for tcms.testplans
         self.assertContains(response, 'Plan types')
 
-        self.assertNotContains(response, 'Test plans')
+        self.assertContains(response, 'Search Test Plans')
```
Comment (atodorov) [inline tcms/core/tests/test_admin.py]: Similar story with "Test plans", the assert matches the navigation menu item, not the admin page. ATM there is a model admin for test plans.

File: tcms/core/tests/test_admin.py [hunk 2]
```diff
@@ -44,21 +43,22 @@ def test_admin_display(self):
...
 
         # for tcms.testruns
-        self.assertNotContains(response, 'Testruns')
```
Comment (atodorov) [inline tcms/core/tests/test_admin.py]: assertContains

File: tcms/core/tests/test_admin.py [hunk 3]
```diff
@@ -44,21 +43,22 @@ def test_admin_display(self):
...
-        self.assertNotContains(response, 'Test case run statuss')
+        self.assertNotContains(response, 'Test case run status')
         self.assertNotContains(response, 'Test case run statuses')
-        self.assertNotContains(response, 'Test runs')
```
Comment (atodorov) [inline tcms/core/tests/test_admin.py]: assertContains

File: tcms/core/tests/test_admin.py [hunk 4]
```diff
@@ -44,21 +43,22 @@ def test_admin_display(self):
...
         self.assertContains(response, 'Plan types')
 
-        self.assertNotContains(response, 'Test plans')
+        self.assertContains(response, 'Search Test Plans')
 
         # for tcms.testruns
-        self.assertNotContains(response, 'Testruns')
-        self.assertNotContains(response, 'Test case run statuss')
+        self.assertNotContains(response, 'Test case run status')
         self.assertNotContains(response, 'Test case run statuses')
```
Comment (atodorov) [inline tcms/core/tests/test_admin.py]: This is now named `TestExecutionStatus` model and it is not currently shown in admin. Once @asankov finishes refactoring of this part we'll have to show these fields again.

Try registering an admin section for the model to see which would be the appropriate name to assert that it is not shown.

File: tcms/core/tests/test_admin.py [hunk 5]
```diff
@@ -44,21 +43,22 @@ def test_admin_display(self):
...
-        self.assertNotContains(response, 'Test runs')
+        self.assertContains(response, 'Search Test Runs')
 
-        # for tcms.xmlrpc
-        self.assertContains(response, 'Xml rpc logs')
```
Comment (atodorov) [inline tcms/core/tests/test_admin.py]: this model has been removed so deletion is correct

File: tcms/core/tests/test_admin.py [hunk 6]
```diff
@@ -44,21 +43,22 @@ def test_admin_display(self):
...
+        # for tcms main modules
+        self.assertContains(response, 'Testruns')
+        self.assertContains(response, 'Testcases')
+        self.assertContains(response, 'Testplans')
```
Comment (atodorov) [inline tcms/core/tests/test_admin.py]: these must be groupped with their corresponding sections above otherwise makes the asserts hard to read

File: tcms/core/tests/test_admin.py [hunk 7]
```diff
@@ -8,7 +8,6 @@
 
 from tcms.tests import BasePlanCase, BaseCaseRun
 from tcms.tests.factories import UserFactory
-from tcms.tests.factories import TestCaseEmailSettingsFactory
 
 
 class TestAdminView(BasePlanCase):
```
Comment (atodorov) [inline tcms/core/tests/test_admin.py]: ATM I don't see a reason why we should inherit from BasePlanCase. Maybe just LoggedInTestCase, no ?

File: tcms/core/tests/test_admin.py [hunk 8]
```diff
@@ -116,14 +118,13 @@ def test_delete_another_user(self):
 
     def test_delete_myself(self):
         self.client.login(  # nosec:B106:hardcoded_password_funcarg
-            username=self.tester.username,
+            username=self.superuser.username,
```
Comment (atodorov) [inline tcms/core/tests/test_admin.py]: Each user should be able to delete themselves so this change is not needed

Comment (atodorov) [reply tcms/core/tests/test_admin.py]: However superuser should be able to delete anyone so that is another test method which is missing

File: tcms/core/tests/test_admin.py [hunk 9]
```diff
@@ -116,14 +118,13 @@ def test_delete_another_user(self):
...
             password='password')
         response = self.client.get(self.url)
 
         # verify there's the Yes, I'm certain button
         self.assertEqual(HTTPStatus.OK, response.status_code)
         self.assertContains(response, "Yes, I'm sure")
-
```
Comment (atodorov) [inline tcms/core/tests/test_admin.py]: unrelated

File: tcms/core/tests/test_admin.py [hunk 10]
```diff
@@ -116,14 +118,13 @@ def test_delete_another_user(self):
...
     def test_delete_myself(self):
         self.client.login(  # nosec:B106:hardcoded_password_funcarg
-            username=self.tester.username,
+            username=self.superuser.username,
             password='password')
         response = self.client.get(self.url)
 
         # verify there's the Yes, I'm certain button
         self.assertEqual(HTTPStatus.OK, response.status_code)
         self.assertContains(response, "Yes, I'm sure")
```
Comment (atodorov) [inline tcms/core/tests/test_admin.py]: this is probably translated in the response and that's why tests in CI with different locales fail. Try marking the string in `_()`

File: tcms/core/tests/test_ajax.py [hunk 11]
```diff
@@ -74,103 +74,29 @@ def test_update_default_tester_non_existing_user(self):
...
-            'usernmae': 'user which doesnt exist'
+            'username': 'user which doesnt exist'
         })
 
-        self.assertEqual(HTTPStatus.OK, response.status_code)
+        self.assertEqual(HTTPStatus.NOT_FOUND, response.status_code)
         result = json.loads(str(response.content, encoding=settings.DEFAULT_CHARSET))
         self.assertEqual(result['rc'], 1)
-        self.assertEqual(result['response'], 'Default tester not found!')
+        self.assertEqual(result['response'], 'User user which doesnt exist not found!')
```
Comment (atodorov) [inline tcms/core/tests/test_ajax.py]: The error message in the view is wrong. It wraps `"User %s not found" % username` into `_()` for translations. The actual string substitution must be done outside of the translation string.

The same substitution should be done here to pass l10n tests in CI.

File: tcms/core/tests/test_ajax.py [hunk 12]
```diff
@@ -193,14 +119,14 @@ def setUpTestData(cls):
     def test_render_plan(self):
         response = self.client.get(self.url, {
             'plan': self.test_plan.plan_id
-        })
+        }, follow=True)
```
Comment (atodorov) [inline tcms/core/tests/test_ajax.py]: Why do you need `follow=True` here? It is usually used when there is a redirect.

File: tcms/core/tests/test_ajax.py [hunk 13]
```diff
@@ -74,103 +74,29 @@ def test_update_default_tester_non_existing_user(self):
...
-            'tags': self.test_tag,
-            'run': self.test_run.run_id,
-            'a': 'remove'
-        })
-
-        self.assertEqual(response.status_code, HTTPStatus.OK)
-        self.assertEqual(self.test_run.tag.count(), 0)
-
-
 class Test_Tag_Render(Test_Tag_Test):
```
Comment (atodorov) [inline tcms/core/tests/test_ajax.py]: You can move the setup from `Test_Tag_Test` here and delete that other class b/c nothing else uses it anymore.

File: tcms/core/tests/test_views.py [hunk 14]
```diff
@@ -44,7 +40,8 @@ def setUpTestData(cls):
         super().setUpTestData()
         # used to reproduce Sentry #KIWI-TCMS-38 where rendering fails
         # with that particular value
-        cls.chinese_tp = TestPlanFactory(name="缺货反馈测试需求",
+        plan_name = u'缺货反馈测试需求'.encode('unicode_escape')
+        cls.chinese_tp = TestPlanFactory(name=plan_name,
```
[the remark this answers is not in the document]
Comment (RMadjev) [reply tcms/core/tests/test_views.py]: MySQLdb._exceptions.OperationalError: (1366, "Incorrect string value: '\\xE7\\xBC\\xBA\\xE8\\xB4\\xA7...' for column 'name' at row 1") and django.db.utils.OperationalError: (1366, "Incorrect string value: '\\xE7\\xBC\\xBA\\xE8\\xB4\\xA7...' for column 'name' at row 1")
in my env.

File: tcms/core/tests/test_admin.py (hunk 2 above)
Comment (RMadjev) [reply tcms/core/tests/test_admin.py]: its add L56-58

File: tcms/core/tests/test_views.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (atodorov) [reply tcms/core/tests/test_views.py]: I guess your DB server is not configured for UTF8, see `.travis/my.cnf`

[the remark this answers is not in the document]
Comment (RMadjev) [reply tcms/core/tests/test_views.py]: Yes, when run it with make test(SQLite) it works. I will revert it.

File: tcms/core/tests/test_ajax.py [hunk 15]
```diff
@@ -71,123 +72,43 @@ def test_update_default_tester_via_email(self):
...
-    @classmethod
-    def setUpTestData(cls):
-        super(Test_Tag_Render, cls).setUpTestData()
-
         cls.test_plan.add_tag(cls.test_tag)
         cls.test_case.add_tag(cls.test_tag)
         cls.test_run.add_tag(cls.test_tag)
 
-        for _ in range(0, 3):
+        for i in range(0, 3):
```
Comment (atodorov) [inline tcms/core/tests/test_ajax.py]: these are unnecessary and cause pylint to report "tcms/core/tests/test_ajax.py:105:12: W0612: Unused variable 'i' (unused-variable)". I will merge this as it is but fix pylint issues reported for these new test files right afterwards.
